common/djangoapps/xblock_django/tests/test_api.py

```diff
@@ -0,0 +1,197 @@
+"""
+Tests related to XBlock support API.
+"""
+from xblock_django.models import XBlockConfiguration, XBlockStudioConfiguration, XBlockStudioConfigurationFlag
+from xblock_django.api import deprecated_xblocks, disabled_xblocks, authorable_xblocks
+from openedx.core.djangolib.testing.utils import CacheIsolationTestCase
+
+
+class XBlockSupportTestCase(CacheIsolationTestCase):
+    """
+    Tests for XBlock Support methods.
+    """
+    def setUp(self):
+        super(XBlockSupportTestCase, self).setUp()
+
+        # Set up XBlockConfigurations for disabled and deprecated states
+        XBlockConfiguration(
+            name="poll",
+            enabled=True,
+            deprecated=True
+        ).save()
```

Comment:
Nit: you might be able to make this a bit more concise by doing something like:

    BLOCK_CONFIG = [
        ("poll", True, True),
        ...
    ]

    for name, enabled, deprecated in BLOCK_CONFIG:
        XBlockConfiguration(name, enabled, deprecated).save()

For the `XBlockStudioConfiguration` the wins would be even more pronounced.